packages/devtools_app/lib/src/screens/inspector_v2/inspector_controller.dart

```diff
@@ -460,6 +515,72 @@ class InspectorController extends DisposableController
     }
   }
 
+  RemoteDiagnosticsNode? _determineNewSelection(
+    RemoteDiagnosticsNode? previousSelection,
+  ) {
+    if (previousSelection == null) return null;
+    if (valueToInspectorTreeNode.containsKey(previousSelection.valueRef)) {
+      return previousSelection;
+    }
+
+    // TODO(https://github.com/flutter/devtools/issues/8481): Consider using a
+    // variation of a path-finding algorithm to determine the new selection,
+    // instead of looking for the first matching descendant.
+    final (closestUnchangedAncestor, distanceToAncestor) =
+        _findClosestUnchangedAncestor(previousSelection);
+    if (closestUnchangedAncestor == null) return inspectorTree.root?.diagnostic;
+
+    final matchingDescendant = _findMatchingDescendant(
+      of: closestUnchangedAncestor,
+      matching: previousSelection,
+      inRange: Range(distanceToAncestor - 3, distanceToAncestor + 3),
+    );
+
+    return matchingDescendant ?? closestUnchangedAncestor;
+  }
+
+  (RemoteDiagnosticsNode?, int) _findClosestUnchangedAncestor(
+    RemoteDiagnosticsNode node, [
+    int distanceToAncestor = 1,
+  ]) {
+    if (valueToInspectorTreeNode.containsKey(node.valueRef)) {
+      final inspectorTreeNode = valueToInspectorTreeNode[node.valueRef];
+      return (inspectorTreeNode?.diagnostic, distanceToAncestor);
+    }
```

Comment:
nit: this can avoid the `containsKey` call by changing to

```
final inspectorTreeNode = valueToInspectorTreeNode[node.valueRef];
if (inspectorTreeNode != null) {
  return (inspectorTreeNode.diagnostic, distanceToAncestor);
}
```

Reply:
oh nice, done!